shell/platform/darwin/macos/framework/Source/FlutterMutatorView.mm

```diff
@@ -219,27 +244,29 @@ - (BOOL)isFlipped {
   return YES;
 }
 
+/// Returns the scale factor to translate logical pixels to physical pixels for this view.
+- (CGFloat)contentsScale {
+  return self.superview != nil ? self.superview.layer.contentsScale : 1.0;
+}
+
 /// Whenever possible view will be clipped using layer bounds.
 /// If clipping to path is needed, CAShapeLayer(s) will be used as mask.
 /// Clipping to round rect only clips to path if round corners are intersected.
 - (void)applyFlutterLayer:(const FlutterLayer*)layer {
-  CGFloat scale = self.superview != nil ? self.superview.layer.contentsScale : 1.0;
-
-  // Initial transform to compensate for scale factor. This is needed because all
-  // cocoa coordinates are logical but Flutter will send the physical to logical
-  // transform in mutations.
-  CATransform3D transform = CATransform3DMakeScale(1.0 / scale, 1.0 / scale, 1);
+  CGFloat scale = [self contentsScale];
+  auto mutations = MutationsForPlatformView(layer->platform_view, scale);
 
   // Platform view transform after applying all transformation mutations.
-  CATransform3D finalTransform = transform;
-  for (size_t i = 0; i < layer->platform_view->mutations_count; ++i) {
-    auto mutation = layer->platform_view->mutations[i];
-    if (mutation->type == kFlutterPlatformViewMutationTypeTransformation) {
-      finalTransform =
-          CATransform3DConcat(ToCATransform3D(mutation->transformation), finalTransform);
+  CATransform3D finalTransform = CATransform3DIdentity;
+  for (auto mutation : mutations) {
+    if (mutation.type == kFlutterPlatformViewMutationTypeTransformation) {
+      CATransform3D mutationTransform = ToCATransform3D(mutation.transformation);
+      finalTransform = CATransform3DConcat(mutationTransform, finalTransform);
```

Comment:
I would have thought this would be in the other order, since the Apple docs say it returns `mutationTransform * finalTransform`, and I would expect that the mutation would be a post-transform (i.e. `finalTransform * mutationTransform`) on the final transform, not a pre-transform. The handedness of the coordinate system can affect that, however, so maybe I'm just mistaken. I think it's supposed to be a right handed coordinate system, though.

Reply:
They're in reverse order. This preserves the existing logic, but just refactors it.